pkg/drivers/unity.go

```diff
@@ -0,0 +1,153 @@
+//  Copyright © 2023 Dell Inc. or its subsidiaries. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//       http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package drivers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	csmv1 "github.com/dell/csm-operator/api/v1"
+	"github.com/dell/csm-operator/pkg/logger"
+	"github.com/dell/csm-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// UnityPluginIdentifier - Unity Plugin Identifier
+	UnityPluginIdentifier = "unity"
+
+	// UnityConfigParamsVolumeMount - Volume mount
+	UnityConfigParamsVolumeMount = "csi-unity-config-params"
+
+	// CsiUnityNodeNamePrefix - Node Name Prefix
+	CsiUnityNodeNamePrefix = "<X_CSI_UNITY_NODENAME_PREFIX>"
+
+	// CsiLogLevel - Defines the log level
+	CsiLogLevel = "<CSI_LOG_LEVEL>"
+
+	// AllowRWOMultipodAccess - Defines if multiple pod should have access to a volume
+	AllowRWOMultipodAccess = "<X_CSI_UNITY_ALLOW_MULTI_POD_ACCESS>"
+
+	// MaxUnityVolumesPerNode - Max volumes per node
+	MaxUnityVolumesPerNode = "<MAX_UNITY_VOLUMES_PER_NODE>"
+
+	// SyncNodeInfoTimeInterval - Interval to sync node info
+	SyncNodeInfoTimeInterval = "<X_CSI_UNITY_SYNC_NODEINFO_INTERVAL>"
+
+	// TenantName - Name of the tenant
+	TenantName = "<TENANT_NAME>"
+)
+
+// PrecheckUnity do input validation
+func PrecheckUnity(ctx context.Context, cr *csmv1.ContainerStorageModule, operatorConfig utils.OperatorConfig, ct client.Client) error {
+	log := logger.GetLogger(ctx)
+	// Check for secret only
+	config := cr.Name + "-creds"
+
+	if cr.Spec.Driver.AuthSecret != "" {
+		config = cr.Spec.Driver.AuthSecret
+	}
+
+	// Check if driver version is supported by doing a stat on a config file
+	configFilePath := fmt.Sprintf("%s/driverconfig/unity/%s/upgrade-path.yaml", operatorConfig.ConfigDirectory, cr.Spec.Driver.ConfigVersion)
+	if _, err := os.Stat(configFilePath); os.IsNotExist(err) {
+		log.Errorw("PreCheckUnity failed in version check", "Error", err.Error())
+		return fmt.Errorf("%s %s not supported", csmv1.Unity, cr.Spec.Driver.ConfigVersion)
+	}
+	secrets := []string{config}
+
```

Comment:
@panigs7  Can we have validations for certs along with the secrets created in the test-unity namespace? Since certs should be created mandatory to install the csi unity driver.

Reply:
@karthikk92 I will take this in another PR. Thanks for pointing it out!!